### PR TITLE
docs: Specify minimum supported rust version

### PR DESCRIPTION
**Pull request by ahmadjubair33**
Updated the rust version in `Cargo.toml` from `1.4.0` to `1.59` .please review this pr and merge it. Thank you
#3716

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "starship"
-version = "1.4.0"
+version = "1.59"
```
Comment (davidkna) [inline Cargo.toml:3]: This sets the version of starship and not the supported rust version. Please also add a comment that the MSRV is intended as a hint and that only the latest version is tested, and that the specified minimum rust version is not guaranteed to work.

Comment (ahmadjubair33) [reply Cargo.toml:3]: Hi @davidkna Can you please the exact location of the rust version where it resides so I can change it.

Comment (davidkna) [reply Cargo.toml:3]: There is no minimum rust version specified in the file. You will need to add it.

Comment (ahmadjubair33) [reply Cargo.toml:3]: In which file?

Comment (davidkna) [reply Cargo.toml:3]: https://github.com/starship/starship/issues/3716 includes a link to the relevant documentation.

Comment (ahmadjubair33) [reply Cargo.toml]: Hi @davidkna I have added the rust version can you please review it.

Comment (davidkna) [reply Cargo.toml]: Please ensure that `Cargo.toml` is still valid TOML, ensure the formatting (dprint, see the contributing guidelines) passes and add the comment I requested.

**Pull request closed without merging**